sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py

```diff
@@ -171,11 +171,15 @@ def _open(self):
 
         auth = None if self._connection else create_authentication(self)
         self._create_handler(auth)
-        self._handler.open(connection=self._connection)
-        self._message_iter = self._handler.receive_messages_iter()
-        while not self._handler.client_ready():
-            time.sleep(0.05)
-        self._running = True
+        try:
+            self._handler.open(connection=self._connection)
+            self._message_iter = self._handler.receive_messages_iter()
+            while not self._handler.client_ready():
+                time.sleep(0.05)
+            self._running = True
+        except:
+            self.close()
```

Comment:
I suggest self._handler.close().
If someday self.close() has more than just self._handler.close(), this might be a problem potentially.

Reply:
This was intentional to include the additional cleanup close() did, such as nulling the handler.

I would have "strong opinions" if close() was not also an idempotent and safe operation in almost all situations, what would you imagine us adding that would render this problematic?

Reply:
Per our OOB discussion, have started the broader discussion over whether we want to allow reopening all up; but with or without that semantic this (keeping open() idempotent) seems prudent; we can add explicit state machine controls to enforce if we want limitations on lifecycle and try to avoid dangling state.